cdap-ui/app/directives/dag/my-dag-ctrl.js

```diff
@@ -432,6 +435,30 @@ angular.module(PKG.name + '.commons')
         vm.instance.repaintEverything();
       });
 
+      NodesStore.registerOnChangeListener(function () {
+        vm.comments = NodesStore.getComments();
+
+        if (!vm.isDisabled) {
+          $timeout(function () {
+            var comments = document.querySelectorAll('.comment-box');
+            vm.instance.draggable(comments, {
+              start: function () {
+                dragged = true;
+              },
+              stop: function (dragEndEvent) {
+                var config = {
+                  _uiPosition: {
+                    top: dragEndEvent.el.style.top,
+                    left: dragEndEvent.el.style.left
+                  }
+                };
+                NodesActionsFactory.updateComment(dragEndEvent.el.id, config);
```

Comment:
Could this cause extra cycles? We update NodeStore and NodeStore emits change and we registerOnChangeListener and we execute the $timeout again. We have been avoiding this pattern because of this reason I believe. If not we should probably re-think updating nodes and connections using this pattern (flux-like way)